include/rocksdb/concurrent_task_limiter.h

```diff
@@ -0,0 +1,45 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include "rocksdb/env.h"
+#include "rocksdb/statistics.h"
+
+namespace rocksdb {
+
+class ConcurrentTaskLimiter {
+ public:
+
+  virtual ~ConcurrentTaskLimiter() {}
+
+  virtual const std::string& GetName() const = 0;
+
+  virtual void SetMaxOutstandingTask(int32_t limit) = 0;
+
+  virtual void ResetMaxOutstandingTask() = 0;
+  
+  virtual int32_t GetOutstandingTask() const = 0;
+
+  virtual bool GetToken(bool force, int32_t& tasks) = 0;
+
+  virtual void ReturnToken(int32_t& tasks) = 0;
+};
+
+// Create a ConcurrentTaskLimiter that can be shared with mulitple CFs
+// across RocksDB instances to control concurrent tasks.
+//
+// @param name: Name of the limiter.
+// @param limit: max concurrent tasks. 
+//        limit = 0 means no new task allowed.
+//        limit < 0 means no limitation.
+extern ConcurrentTaskLimiter* NewConcurrentTaskLimiter(
+    const std::string& name = "", int32_t limit = -1);
```

Comment:
Do we need default values for them? Most of the default values were added for backward compatibility reason.
This usage doesn't seem to follow what Google C++ Style's recommandation: https://google.github.io/styleguide/cppguide.html#Default_Arguments

Reply:
fixed.